src/java.desktop/share/classes/java/awt/TextArea.java

```diff
@@ -606,6 +607,15 @@ public Dimension minimumSize() {
         }
     }
 
+    @Override
+    public synchronized void setEditable(boolean b) {
+        super.setEditable(b);
+        Color defaultBackground = UIManager.getColor("TextArea.background");
```

Comment:
I am not sure if it is ok to use swing "UIManager" class in awt
I guess long time back somebody maybe @prrace mentioned that we should avoid calling swing class from awt? 
I see we use AttributeSet in TextComponent for accessibility usage but am not sure if we can use without restriction?

Reply:
This call is only to get the default background of TextArea. Is there another way to grab the default color without using UIManager?

Reply:
Yes, but as far I could see `TextArea.background` property is being set in 
`AquaLookAndFeel.java, MotifLookAndFeel.java, BasicLookAndFeel.java, XAWTLookAndFeel.java and WindowsLookAndFeel.java`
and this L&F will not be set in AWT environment, if I am not wrong so you will either get garbage or wrong default color if you use UIManager.getColor..so I guess you need to use `TextComponent.getBackground()` which returns SystemColor.control if nothing is set

Reply:
i've made the change to use TextComponent.getBackground() instead to get the correct background color